pandas/io/pytables.py

```diff
@@ -691,41 +697,14 @@ def open(self, mode: str = "a", **kwargs):
                 self._complevel, self._complib, fletcher32=self._fletcher32
             )
 
-        try:
-            self._handle = tables.open_file(self._path, self._mode, **kwargs)
-        except OSError as err:  # pragma: no cover
-            if "can not be written" in str(err):
-                print(f"Opening {self._path} in read-only mode")
-                self._handle = tables.open_file(self._path, "r", **kwargs)
-            else:
-                raise
```

Comment:
I checked the current code of ``tables.open_file`` and did not find error messages containing ``can not be written``. Even in 3.1 there is no such message. Considering the current minimal pytables version (>= 3.4.3), this error handing seems to be irrelevant (``if "can not be written" in str(err)``).

Reply:
yeah this could have changed, was hard to test :->